components/style/values/specified/list.rs

```diff
@@ -73,28 +74,17 @@ impl Parse for ListStyleType {
     }
 }
 
-/// Specified and computed `list-style-image` property.
-#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToCss)]
-pub struct ListStyleImage(pub ImageUrlOrNone);
-
-// FIXME(nox): This is wrong, there are different types for specified
-// and computed URLs in Servo.
-trivial_to_computed_value!(ListStyleImage);
-
-impl ListStyleImage {
-    /// Initial specified value for `list-style-image`.
-    #[inline]
-    pub fn none() -> ListStyleImage {
-        ListStyleImage(Either::Second(None_))
-    }
-}
+/// Specified `list-style-image` property.
+pub type ListStyleImage = GenericListStyleImage<SpecifiedUrl>;
 
 impl Parse for ListStyleImage {
     fn parse<'i, 't>(
         context: &ParserContext,
         input: &mut Parser<'i, 't>,
     ) -> Result<ListStyleImage, ParseError<'i>> {
-        ImageUrlOrNone::parse(context, input).map(ListStyleImage)
+        SpecifiedUrl::parse(context, input).map(|css_url| {
+            GenericListStyleImage(ImageUrlOrNone::Url(css_url))
```

Comment:
This is wrong. Needs to handle parsing `none` too.

Reply:
Can I just match on the input with `"none"` and `_` as match arms?

Then `"none"` would just return the `::None` variant and `_` would attempt the current implementation.

Reply:
No, because it needs to be an identifier token and all that, and it needs to be ascii-case-insensitive.

You should implement `parse` for `specified::ImageUrlOrNone` as follows:

```rust
if let Ok(url) = input.try(|input| SpecifiedUrl::parse(context, input)) {
    return Ok(ImageUrlOrNone::Url(url));
}
input.expect_ident_matching("none")?;
ImageUrlOrNone::None
```

With that, `ListStyleImage` becomes:

```rust
Ok(GenericListStyleImage(ImageUrlOrNone::parse(context, input)?))
```